### DISCUSSION TITLE
Sticky key modifier

### DISCUSSION THREAD

**Discussion by otherJL0** [ghostty-org/ghostty#3737] [Feature Requests, Ideas] [answered]
I think the keybinds can be extended with a concept of "sticky" keys for trigger sequences by holding the modifier key of the second key combination
Examples:

```
# after pressing ctrl-a, the user can hold shift
# to continuously switch to the next tab
# until the user lets go of shift
keybind = ctrl+a>shift-n=next_tab
keybind = ctrl+a>shift-h=resize_split:10,left
```

This might require new syntax with some ideas below:
```
# prefix sticky modifier with `sticky`
keybind = ctrl-a>sticky:shift+n=next_tab

# modifier must be in all caps to indicate this is sticky
keybind = ctrl-a>SHIFT+x=close_surface

# use a new operator `++` to indicate stickiness 
keybind = ctrl-a>shift++t=move_tab:-1
```

**Comment by rhodes-b** [comment, marked as answer]
This sounds like [trigger sequences](https://ghostty.org/docs/config/keybind/sequence) so I believe ghostty supports this

now if you are also going to ask for a delay similar to tmux there is another discussion for that https://github.com/ghostty-org/ghostty/discussions/4659
